src/IO/ParallelReadBuffer.cpp

```diff
@@ -278,7 +277,6 @@ void ParallelReadBuffer::readerThreadFunction(ReadWorkerPtr read_worker)
 
 void ParallelReadBuffer::onBackgroundException()
 {
-    std::lock_guard lock(mutex);
```

Comment:
why we don't need synchronization among all workers here?

Reply:
nice catch! we should sync